### PR TITLE
Add sync account feature

### LINKED ISSUE #14: [Account Sync] Syncing screen needs an activity indicator
When I start syncing my account, I see the loading status as 0%, and then it is immediately completed without seeing any other indicators. This almost felt as if it froze. Instead of the 0%, we should show a loading indicator and when the percentage is > 0%, then replace the loading icon with the percent loaded.

### LINKED ISSUE #55: [Account Sync] Onboarding - Sync account
- [x] Option to scan a QR code
- [x] Option to enter a public key manually
- [x] Option to recover from cloud backup (I'm not logged in on any other devices)

Note:

### PR DESCRIPTION
**Pull request by vitalikgriza**
Implement sync account feature

### CODE REVIEW

File: src/screens/SyncSuccess/index.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      <Text style={fonts.title2}>Account Synced</Text>
+      <Text style={[fonts.regular.regular, { marginTop: spacing.sm }]}>
+        Success!
+      </Text>
+      <Text style={[fonts.regular.regular, { marginTop: spacing.md }]}>
+        You can now log in with your synced account.
+      </Text>
+      <View style={styles.imageContainer}>
+        <Image
+          source={require('../../assets/images/syncSuccess.png')}
```
Comment (suleymanCel) [inline src/screens/SyncSuccess/index.tsx]: we can add this image to our common file for index 
import images from '../../../assets/images';
exp use: 
 <Image
    source={ images.syncSuccess}
    ..
  />

File: src/screens/SyncSuccess/index.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    } catch (error) {
+      Alert.alert('Error', 'Error on logging in.');
+      setIsLoading(false);
+    }
+  };
+
+  return (
+    <ScrollableContainer>
+      <Text style={fonts.title2}>Account Synced</Text>
+      <Text style={[fonts.regular.regular, { marginTop: spacing.sm }]}>
```
Comment (suleymanCel) [inline src/screens/SyncSuccess/index.tsx]: inline style, we can move this to the style file

File: src/screens/SyncSuccess/index.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  const onPressContinue = async () => {
+    try {
+      setIsLoading(true);
+      // we need to logout before we login again
+      await dispatch(accountLogout());
+      // navigate automatically to main screen
+      await dispatch(loginFlow({ email, password: masterPassword })).unwrap();
+    } catch (error) {
+      Alert.alert('Error', 'Error on logging in.');
```
Comment (suleymanCel) [inline src/screens/SyncSuccess/index.tsx]: maybe, toast

File: src/screens/SyncPending/index.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    filesSynced > 0 && filesSynced <= 100 ? (
+      <Text style={styles.btnText}>{`${filesSynced}%`}</Text>
+    ) : (
+      <ActivityIndicator />
+    );
+
+  return (
+    <ScrollableContainer>
+      <Text style={fonts.title2}>Syncing</Text>
+      <Text style={[fonts.regular.regular, { marginTop: spacing.sm }]}>
```
Comment (suleymanCel) [inline src/screens/SyncPending/index.tsx]: inline style, we can move this to the style file

File: src/screens/SyncPending/index.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  const { syncData, masterPassword } = route.params;
+  const onSyncSuccess = () => {
+    navigation.navigate('syncSuccess', {
+      email: syncData.email,
+      masterPassword,
+    });
+  };
+
+  const onSyncError = (error: any) => {
+    Alert.alert('Error', 'Error on syncing account.' + error.message);
```
Comment (suleymanCel) [inline src/screens/SyncPending/index.tsx]: maybe we can use react-native-toast-message

File: src/screens/SyncNewDevice/index.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    }
+  };
+
+  useEffect(() => {
+    generateSyncInfo();
+  }, [showCode, showQrCode]);
+
+  return (
+    <ScrollableContainer>
+      <Text style={[fonts.regular.regular, { marginTop: spacing.sm }]}>
```
Comment (suleymanCel) [inline src/screens/SyncNewDevice/index.tsx]: inline style, we can move this to the style file

File: src/screens/SyncMasterPassword/index.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+import { Input } from '../../components/Input';
+
+type SyncMasterPasswordScreenProps = NativeStackScreenProps<
+  SyncStackParams,
+  'syncMasterPassword'
+>;
+
+export default ({ navigation, route }: SyncMasterPasswordScreenProps) => {
+  const { syncData } = route.params;
+  const [password, setPassword] = useState('');
```
Comment (suleymanCel) [inline src/screens/SyncMasterPassword/index.tsx:18]: not required but maybe we can add type

File: src/screens/SyncNewDevice/index.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+  };
+
+  const generateSyncInfo = async () => {
+    if (showCode || (showQrCode && !syncInfo?.email && !syncInfo?.drive_key)) {
+      try {
+        setIsLoading(true);
+        const resp = await dispatch(createAccountSyncInfo()).unwrap();
+        setSyncInfo(resp);
+      } catch (error) {
+        Alert.alert(
```
Comment (suleymanCel) [inline src/screens/SyncNewDevice/index.tsx]: If doesn't required, maybe we can use react-native-toast-message as other screens.

File: src/screens/SyncMasterPassword/index.tsx (hunk 7 above)
Comment (vitalikgriza) [reply src/screens/SyncMasterPassword/index.tsx:18]: There is no need of typing, as it's automatically cast to string

**Review by suleymanCel — APPROVED** [review]

**Pull request merged**